docs/src/modules/concepts/pages/architecture-model.adoc

```diff
@@ -1,132 +1,81 @@
 
-image::docs-onion_architecture-v1.min.svg[Layered Structure of a Service, width=400]
-
-Each layer should live in its own package:
-
-* `domain`: for the domain model
-* `application`: for Akka components
-* `api`: for external-facing endpoints
-
-Try not to expose domain types directly to the outside world. The API layer should not call the domain layer. Inner layers should not depend on or be aware of outer layers.
-
-== Layer overview
-
-=== Domain
-
-This layer contains business rules and domain concepts. It does not depend on Akka or other runtime concerns. These are plain Java classes, often using `record` to reduce boilerplate. Examples include logic to enforce limits, compute totals, or apply rules.
-
-You can write unit tests for this layer without needing to start Akka or the runtime. The domain package remains isolated, focused, and easy to change.
-
-=== Application
-
-This layer connects the domain model to the Akka runtime. It contains the components that handle persistence, coordination, and external interaction. These components follow event-driven patterns and manage state in a way that supports consistency and responsiveness.
-
-Most classes in this layer are based on Akka-provided building blocks. The domain logic remains in the inner layer. This layer makes it operational.
-
-=== API
-
-This layer connects your service to the outside world. It defines endpoints that expose application functionality over HTTP or gRPC. Requests are handled here and passed on to the application layer.
-
-Endpoints use xref:java:component-and-service-calls.adoc#_component_client[`ComponentClient`] to call Akka components in the application layer. This maintains separation of concerns and ensures runtime boundaries are respected.
-
-The API layer may also expose public event models over Kafka or other channels. External systems should interact with your service only through this layer.
-
-Access control and request validation also belong here. For HTTP-specific guidance, see xref:java:http-endpoints.adoc[Designing HTTP Endpoints].
-
-== Mapping layers to project structure
-
-Each layer described above corresponds to a distinct package in the source tree. This structure helps reinforce separation of concerns and makes it easy to locate different types of logic in a project.
-
-A typical Akka service might have a layout like the following:
+A typical Akka project might have a layout like the following:
 
 [source,txt]
 ----
 src/
- └── main/
-     └── java/acme/petclinic/
-         ├── domain/        # Business logic
-         ├── application/   # Akka components
-         └── api/           # External endpoints
+ ├── main/
+ │   ├── java/acme/planningagent/
+ │   │   ├── api/           # External MCP, HTTP, gRPC endpoints
+ │   │   ├── application/   # Akka components: Agents, Entities, Views, Consumers, Timers, Workflows
+ │   │   ├── domain/        # Business logic
+ │   │   └── tool/          # MCP or functional tools
```

Comment:
tool package? I think tools should be in the application package. Inside the application, you can organize stuff however you want.